xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp

```diff
@@ -618,6 +618,18 @@ bool CAESinkALSA::Initialize(AEAudioFormat &format, std::string &device)
   }
   // adjust format to the configuration we got
   format.m_channelLayout = GetChannelLayout(format, outconfig.channels);
+  // we might end up with an unusable channel layout that only contains
+  // UNKNOWN1 channels, let's do a sanity check
+  bool isSaneLayout = false;
+  for (unsigned int i = 0; i < format.m_channelLayout.Count(); ++i)
+  {
+    isSaneLayout = (format.m_channelLayout[i] != AE_CH_UNKNOWN1);
```

Comment:
Code wise it's not done quite well. Implementing an IsUnknown should be done in AEChannelData, but this one is only used in private API. Adding it to AEChannelInfo somehow sucks - but let's try that.